Server/JokesOnYou.Web.Api/Services/JokesService.cs

```diff
@@ -0,0 +1,32 @@
+﻿using AutoMapper;
+using JokesOnYou.Web.Api.DTOs;
+using JokesOnYou.Web.Api.Repositories.Interfaces;
+using JokesOnYou.Web.Api.Services.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace JokesOnYou.Web.Api.Services
+{
+    public class JokesService : IJokesService
+    {
+        private readonly IJokesRepository _jokesRepo;
+        private readonly IMapper _mapper;
+
+        public JokesService(IJokesRepository jokesRepo, IMapper mapper)
+        {
+            _jokesRepo = jokesRepo;
+            _mapper = mapper;
+        }
+
+        public async Task<IEnumerable<JokeReplyDto>> GetAllJokesAsync()
+        {
+            var jokes = await _jokesRepo.GetAllJokesAsync();
+
```

Comment:
Probably the best way to handle this is to use automapper https://docs.automapper.org/en/stable/Queryable-Extensions.html. This way we only SELECT the properties we need from the db and we dont even need to manually include navigation props.

Reply:
oh cool thanks! i check this out tonight!

Reply:
Very nice this works. exactly what I was worried about ;)
I tried different as well but that didn't work.
I tried 
```
var jokes = await _context.Jokes.Include( joke => joke.Author.Id)
                                .Include(joke => joke.Author.UserName)
                                .ToListAsync();
```
Must be a way to get this to work? but maybe not.